drivers/hud_apr/app/controllers/hud_apr/questions_controller.rb

```diff
@@ -26,5 +27,14 @@ def create
     private def set_question
       @question = generator.valid_question_number(params[:question] || params[:id])
     end
+
+    private def column_headings
+      return HudApr::Fy2020::AprClient.detail_headers unless question_fields(@question).present?
+
+      question_fields(@question).map do |key|
+        [key, HudApr::Fy2020::AprClient.detail_headers[key.to_s]]
+      end.to_h
+    end
+    helper_method :column_headings
```

Comment:
This should all be moved to the question models (or their concerns)

Reply:
✅